apps/vendor/models.py

```diff
@@ -0,0 +1,168 @@
+from django.db import models
+import os
+from uuid import uuid4
+from apps.customer.models import LedgerAccounts
+from apps.masters.models import FirmStatuses, GstCategories, PriceCategories, Territory, Transporters
+
+# Create your models here.
+    
+class VendorCategory(models.Model):
+    vendor_category_id = models.AutoField(primary_key=True)
+    code = models.CharField(max_length=50, null=True, default=None)
+    name = models.CharField(max_length=255, null=True, default=None)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f'{self.vendor_category_id}_{self.name} ({self.code})'
+
+    class Meta:
+        db_table = 'vendor_category'
+
+
+class VendorPaymentTerms(models.Model):
+    payment_term_id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=255, null=True, default=None)
+    code = models.CharField(max_length=50, null=True, default=None)
+    fixed_days = models.IntegerField(null=True, default=None)
+    no_of_fixed_days = models.IntegerField(null=True, default=None)
+    payment_cycle = models.CharField(max_length=255, null=True, default=None)
+    run_on = models.CharField(max_length=255, null=True, default=None)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f'{self.payment_term_id}_{self.name} ({self.code})'
+    
+    class Meta:
+        db_table = 'vendor_payment_terms'
+
+
+class VendorAgent(models.Model):
+    vendor_agent_id = models.AutoField(primary_key=True)
+    code = models.CharField(max_length=50, null=True, default=None)
+    name = models.CharField(max_length=255, null=True, default=None)
+    commission_rate = models.DecimalField(max_digits=18, decimal_places=2, null=True, default=None)
+    
+    RATE_ON_CHOICES = [
+        ('Qty', 'Quantity'),
+        ('Amount', 'Amount'),
+    ]
+    rate_on = models.CharField(max_length=20, choices=RATE_ON_CHOICES, null=True, default=None)
+    
+    AMOUNT_TYPE_CHOICES = [
+        ('Taxable', 'Taxable'),
+        ('BillAmount', 'Bill Amount'),
+    ]
+    amount_type = models.CharField(max_length=20, choices=AMOUNT_TYPE_CHOICES, null=True, default=None)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f'{self.vendor_agent_id}_{self.name} ({self.code})'
+    
+    class Meta:
+        db_table = 'vendor_agent'
+
+
+def custom_upload_to(instance, filename):
+    file_extension = filename.split('.')[-1]
```

Comment:
Why this function is outside the class? if it is common function then move it to utils_methods.py

Reply:
I move Thulasi.